pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -47,25 +48,31 @@ func NewInstanceTypeProvider(ec2api ec2iface.EC2API) *InstanceTypeProvider {
 }
 
 // Get instance types that are availble per availability zone
-func (p *InstanceTypeProvider) Get(ctx context.Context, zonalSubnetOptions map[string][]*ec2.Subnet, constraints Constraints) ([]*packing.Instance, error) {
+func (p *InstanceTypeProvider) Get(ctx context.Context, zonalSubnetOptions map[string][]*ec2.Subnet, constraints Constraints) ([]cloudprovider.InstanceType, error) {
 	zones := []string{}
 	for zone := range zonalSubnetOptions {
 		zones = append(zones, zone)
 	}
 
-	var supportedInstanceTypes []*packing.Instance
-	if instanceTypes, ok := p.cache.Get(allInstanceTypesKey); ok {
-		supportedInstanceTypes = instanceTypes.([]*packing.Instance)
+	var instanceTypes []*InstanceType
+	if cached, ok := p.cache.Get(allInstanceTypesKey); ok {
+		instanceTypes = cached.([]*InstanceType)
 	} else {
 		var err error
-		supportedInstanceTypes, err = p.getZonalInstanceTypes(ctx)
+		instanceTypes, err = p.getZonalInstanceTypes(ctx)
 		if err != nil {
 			return nil, err
 		}
-		p.cache.SetDefault(allInstanceTypesKey, supportedInstanceTypes)
-		zap.S().Debugf("Successfully discovered %d EC2 instance types", len(supportedInstanceTypes))
+		p.cache.SetDefault(allInstanceTypesKey, instanceTypes)
+		zap.S().Debugf("Successfully discovered %d EC2 instance types", len(instanceTypes))
 	}
-	return p.filterFrom(supportedInstanceTypes, constraints, zones), nil
+
+	// Filter by constraints and zones
```

Comment:
Could you change this comment to explain why a copy is being made in this fashion? (I'm not quite sure why not just return the result of `filterFrom` directly? (If it's just to convert from `*InstanceType` to `InstanceType`, why not just have `Get` return a pointer as it used to?) (Or change `filterFrom` not to return pointers?)

Reply:
We're transforming from a map to a slice. In general, I'd like to make this simply not a map (I don't think this needs to be), but I'm addressing in a followon PR.

Reply:
👍🏻

Reply:
Where are we transforming from a map to a slice?  `filterFrom` returns a slice

Reply:
Ah -- sorry you're right. FilterFrom is returning an aws.InstanceType (struct) and we need to convert to cloudprovider.InstanceType (interface). Annoying go inheritance things =/. 

```
cannot use p.filterFrom(instanceTypes, constraints, zones) (value of type []*InstanceType) as []cloudprovider.InstanceType value in return statement
```

Reply:
Added a comment here.